modules/javafx.graphics/src/main/java/com/sun/prism/d3d/D3DTextureData.java

```diff
@@ -36,14 +36,13 @@ public class D3DTextureData extends D3DResource.D3DRecord {
     static long estimateSize(int physicalWidth, int physicalHeight,
                              PixelFormat format)
     {
-        return ((long) physicalWidth) * ((long) physicalHeight) *
-               ((long) format.getBytesPerPixelUnit());
+        return (long) physicalWidth * physicalHeight * format.getBytesPerPixelUnit();
     }
 
     static long estimateRTSize(int physicalWidth, int physicalHeight,
                                boolean hasDepth)
     {
-        return ((long) physicalWidth) * ((long) physicalHeight) * 4L;
+        return (long) physicalWidth * physicalHeight * 4L;
```

Comment:
Another case where the old code might be clearer.

Reply:
good catch - I think the change is not equivalent.  It should be either
```
4L * physicalWidth * physicalHeight;
```

or

```
((long)physicalWidth) * physicalHeight * 4L;
```

the proposed change might result in overflow bc the first intermediate result is likely to be int.

Reply:
Actually, I think the new code correct, though less clear, since `(long)` is applied to `physicalWidth` before the multiplication. The fact that we had to scratch our head and discuss it highlights the point I was trying to make. I do think this is one place where reverting the code back would be best. Then it's obvious that all of the intermediate calculations are in long.

Reply:
The cast has precedence and Java will not downcast mid calculation and lose precision, so all examples here are correct.

Reply:
That variant is fine with me.

Reply:
Updated with this variant.